pkg/registry/apis/provisioning/jobs/export/worker.go

```diff
@@ -56,187 +47,45 @@ func (r *ExportWorker) Process(ctx context.Context, repo repository.Repository,
-	}
-	err = folderTree.Walk(ctx, func(ctx context.Context, folder resources.Folder) error {
-		p := folder.Path + "/"
-		if options.Prefix != "" {
-			p = options.Prefix + "/" + p
-		}
-		logger := logger.With("path", p)
-
-		_, err = repo.Read(ctx, p, ref)
-		if err != nil && !(errors.Is(err, repository.ErrFileNotFound) || apierrors.IsNotFound(err)) {
-			logger.Error("failed to check if folder exists before writing", "error", err)
-			return fmt.Errorf("failed to check if folder exists before writing: %w", err)
-		} else if err == nil {
-			logger.Info("folder already exists")
-			summary.Noop++
-			return nil
-		}
-
-		// Create with an empty body will make a folder (or .keep file if unsupported)
-		if err := repo.Create(ctx, p, ref, nil, "export folder `"+p+"`"); err != nil {
-			logger.Error("failed to write a folder in repository", "error", err)
-			return fmt.Errorf("failed to write folder in repo: %w", err)
-		}
-		summary.Create++
-		logger.Debug("successfully exported folder")
-		return nil
-	})
+	// Load and write all folders
+	err = worker.loadFolders(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("failed to write folders: %w", err)
+		return worker.jobStatus, err
 	}
-	status.Summary = append(status.Summary, summary)
-	status.Message = "writing dashboards..."
 
-	if err := progress(ctx, status); err != nil {
-		return nil, err
-	}
-
-	// TODO: handle pagination
-	dashboardList, err := dashboardsClient.List(ctx, metav1.ListOptions{Limit: 1000})
-	if err != nil {
-		return nil, fmt.Errorf("failed to list dashboards: %w", err)
-	}
-
-	commitMessage := fmt.Sprintf("grafana export to: %s", repo.Config().Name)
-	summary = provisioning.JobResourceSummary{
-		Group:    "dashboard.grafana.app",
-		Resource: "dashboards",
-	}
-	for _, item := range dashboardList.Items {
-		if ctx.Err() != nil {
-			logger.Debug("cancelling replication process due to ctx error", "error", err)
-			return nil, ctx.Err()
-		}
-
-		name := item.GetName()
-		repoName := item.GetAnnotations()[apiutils.AnnoKeyRepoName]
-		if repoName == repo.Config().GetName() {
-			logger.Info("skip dashboard since it is already in repository", "dashboard", name)
-			continue
-		}
-
-		title, _, _ := unstructured.NestedString(item.Object, "spec", "title")
-		if title == "" {
-			title = name
-		}
-		folder := item.GetAnnotations()[apiutils.AnnoKeyFolder]
-
-		// Get the absolute path of the folder
-		fid, ok := folderTree.DirPath(folder, "")
-		if !ok {
-			logger.Error("folder of item was not in tree of repository")
-			return nil, fmt.Errorf("folder of item was not in tree of repository")
-		}
-
-		delete(item.Object, "metadata")
-		if options.Identifier {
-			item.SetName(name) // keep the identifier in the metadata
-		}
-
-		body, err := json.MarshalIndent(item.Object, "", "  ")
+	// Add dashboards, library panels and playlists
+	kinds := []schema.GroupVersionResource{{
+		Group:    dashboards.GROUP,
+		Version:  "v1alpha1",
+		Resource: dashboards.DASHBOARD_RESOURCE,
+	}, {
```

Comment:
we weren't planning to support these 2 for the prototype, weren't we? I think we should stick to dashboards and folders if that's the case.

Reply:
I don't feel strongly -- but I do feel strongly that we must write it so that adding more things is trivial not "hard" 

In particular -- we need an ID/folder/path strategy that does not break when we add things that are not allowed to be in a folder tree :)